cupy/cuda/stream.pyx

```diff
@@ -291,23 +301,36 @@ class Stream(BaseStream):
 
     Args:
         null (bool): If ``True``, the stream is a null stream (i.e. the default
-            stream that synchronizes with all streams). Otherwise, a plain new
-            stream is created. Note that you can also use ``Stream.null``
-            singleton object instead of creating new null stream object.
-        non_blocking (bool): If ``True``, the stream does not synchronize with
-            the NULL stream.
+            stream that synchronizes with all streams). Note that you can also
+            use the ``Stream.null`` singleton object instead of creating a new
+            null stream object.
+        ptds (bool): If ``True`` and ``null`` is ``False``, the per-thread
+            default stream is used. Note that you can also use the
+            ``Stream.ptds`` singleton object instead of creating a new
+            per-thread default stream object.
+        non_blocking (bool): If ``True`` and both ``null`` and ``ptds`` are
+            ``False``, the stream does not synchronize with the NULL stream.
+
+        Note that if both ``null`` and ``ptds`` are ``False``, a plain new
+        stream is created.
 
     Attributes:
         ~Stream.ptr (intptr_t): Raw stream handle. It can be passed to
             the CUDA Runtime API via ctypes.
 
     """
 
-    def __init__(self, null=False, non_blocking=False):
+    def __init__(self, null=False, non_blocking=False, ptds=False):
         if null:
-            self.ptr = 0
+            # TODO(pentschev): move to streamLegacy. This wasn't possible
+            # because of a NCCL bug that should be fixed in the version
+            # following 2.8.3-1.
+            self.ptr = runtime.streamDefault
```

Comment:
nitpick: `streamDefault` is a flag, not a stream macro
```suggestion
            self.ptr = 0
```

Reply:
Stop using my own arguments from https://github.com/cupy/cupy/pull/4322#discussion_r530302943 against myself! 🤣

Reply:
Ah ha ha, forgot about that one! 😄

Reply:
That's why you shouldn't write things that can be used against yourself, I trapped myself with that one! 😄